h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -205,10 +225,27 @@ boolean isUsefulBaseModel(Key<Model> baseModelKey) {
   }
 
   ModelMetrics doScoreMetricsOneFrame(Frame frame, Job job) {
+    if (_parms._training_scoring_subsample_size > 0 && _parms._training_scoring_subsample_size < frame.numRows()) { 
+Frame scoredFrame = (_parms._training_scoring_subsample_size > 0 && _parms._training_scoring_subsample_size < frame.numRows()) 
+        ? MRUtils.sampleFrame(frame, _parms._training_scoring_subsample_size, _parms._seed)
+        : frame;
+try {
+      Frame pred = this.predictScoreImpl(frame, new Frame(scoredFrame), null, job, true, CFuncRef.from(_parms._custom_metric_func));
+      pred.delete();
+      return ModelMetrics.getFromDKV(this, frame);
+} finally {
+     if (scoredFrame != frame) scoredFrame.delete();
+}  
+      Frame pred = this.predictScoreImpl(frame, new Frame(frame), null, job, true, CFuncRef.from(_parms._custom_metric_func));
+      pred.delete();
+      ModelMetrics metrics = ModelMetrics.getFromDKV(this, frame);
+      frame.delete();
+      return metrics;
+    } else {
       Frame pred = this.predictScoreImpl(frame, new Frame(frame), null, job, true, CFuncRef.from(_parms._custom_metric_func));
-//      Frame pred = this.score(frame, null, job, true,  CFuncRef.from(_parms._custom_metric_func));
       pred.delete();
       return ModelMetrics.getFromDKV(this, frame);
+    }
```

Comment:
suggestion was committed in wrong place apparently... it couldn't be applied to whole block due to deleted lines, and beside I forgot to rename some `frame` into `scoreFrame`, should be sth like.
```
Frame scoredFrame = (_parms._training_scoring_subsample_size > 0 && _parms._training_scoring_subsample_size < frame.numRows()) 
        ? MRUtils.sampleFrame(frame, _parms._training_scoring_subsample_size, _parms._seed)
        : frame;
try {
      Frame pred = this.predictScoreImpl(scoredFrame, new Frame(scoredFrame), null, job, true, CFuncRef.from(_parms._custom_metric_func));
      pred.delete();
      return ModelMetrics.getFromDKV(this, scoredFrame);
} finally {
     if (scoredFrame != frame) scoredFrame.delete();
}  
```
better apply it locally first though.

Reply:
Thanks!